src/user/user.service.ts

```diff
@@ -2,17 +2,16 @@ import { Injectable } from '@nestjs/common'
 import { InjectRepository } from '@nestjs/typeorm'
 import { User } from '../entities/user.entity'
 import { FindConditions, FindOneOptions, Repository } from 'typeorm'
-import { UpdateRelationshipDto, CreateUserDto, GetUserDto } from './dto/user.dto'
+import { UpdateRelationshipDto, CreateUserDto, IDto } from './dto/user.dto'
 import { DuplicateElementException, InvalidUserTypeException, UserNotFoundException } from './user.exception'
+import { plainToInstance } from 'class-transformer'
 
 @Injectable()
 export class UserService {
+  private salt = 10
```

Comment:
In my opinion salt should be in the env not in the source code and we should also keep that in secret channel